packages/trezor/src/trezor.ts

```diff
@@ -0,0 +1,233 @@
+      const coin = chain.toLowerCase() as 'btc' | 'bch' | 'ltc' | 'doge';
+      const toolbox = (
+        chain === Chain.Bitcoin ? BTCToolbox : chain === Chain.Litecoin ? LTCToolbox : DOGEToolbox
+      )(utxoApiKey);
+
+      const signTransaction = async (psbt: Psbt, memo: string = '') => {
+        //@ts-ignore
+        const result = await TrezorConnect.signTransaction({
+          coin,
+          inputs: psbt.txInputs.map((input) => ({
+            address_n: derivationPath.map((pathElement, index) =>
+              index < 3 ? (pathElement | 0x80000000) >>> 0 : pathElement,
+            ),
+            prev_hash: input.hash.reverse().toString('hex'),
+            prev_index: input.index,
+            // object needs amount but does not use it for signing
+            amount: 0,
+          })),
+          outputs: psbt.txOutputs.map((output) => {
+            if (!output.address) {
+              return {
+                op_return_data: Buffer.from(memo).toString('hex'),
+                amount: '0',
+                script_type: 'PAYTOOPRETURN',
+              };
+            }
+            return {
+              address: output.address as string,
+              amount: output.value,
+              script_type: 'PAYTOADDRESS',
+            };
+          }),
+        });
+
+        if (result.success) {
+          return result.payload.serializedTx;
+        } else {
+          throw new Error(
+            `Trezor failed to sign the ${chain.toUpperCase()} transaction: ${
+              (result.payload as { error: string; code?: string }).error
+            }`,
+          );
+        }
+      };
+
+      const transfer = async ({
+        from,
+        recipient,
+        feeOptionKey,
+        feeRate,
+        memo,
+        ...rest
+      }: UTXOTransferParams) => {
+        if (!from) throw new Error('From address must be provided');
+        if (!recipient) throw new Error('Recipient address must be provided');
+        const { psbt } = await toolbox.buildTx({
+          ...rest,
+          memo,
+          feeOptionKey,
+          recipient,
+          feeRate: feeRate || (await toolbox.getFeeRates())[feeOptionKey || FeeOption.Fast],
+          sender: from,
+          fetchTxHex: chain === Chain.Doge,
+        });
+
+        const txHex = await signTransaction(psbt, memo);
+        return toolbox.broadcastTx({ txHex });
+      };
+
+      const getAddress = async (path: DerivationPathArray = derivationPath) => {
+        //@ts-ignore
+        const { success, payload } = await TrezorConnect.getAddress({
+          path: `m/${derivationPathToString(path)}`,
+          coin,
+        });
+
+        if (!success)
+          throw new Error(
+            'Failed to get address: ' +
+              ((payload as { error: string; code?: string }).error || 'Unknown error'),
+          );
+
+        return payload.address;
+      };
+
+      const address = await getAddress();
```

Comment:
This feels unnecessary -> we should either extract functions is possible outside of it or just use directly form `TrezorConnect.getAddress` call